src/file.ts

```diff
@@ -46,6 +46,22 @@ function withMimeType(file: File) {
 
 function clone(file: File, type?: string) {
     const data = file.slice();
+
+    // Edge Hack
+    // ---------
+    // Edge doesn't support File constructor so instead we're using Blob and filling missing fields so it
+    // will look like a File
+    if (typeof window.navigator.msSaveBlob !== 'undefined') {
```

Comment:
```suggestion
    if (!('File' in window) && 'Blob' in window) {
```

Reply:
This test will fail because Edge and IE actually have File class.
![edge test](https://user-images.githubusercontent.com/2941275/48936465-17aaea80-ef1c-11e8-8098-b95cfbfc9578.PNG)

Reply:
I'm confused, I thought that was the issue, that the ctor did not exist. Because the check for `msSaveBlob` is misleading as that method's purpose is to save a File or Blob to disk.

Reply:
I'd use [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/construct) and throw when the handler is invoked.

Reply:
@rolandjitsu I mean, I don't know how to replace File class with my mock.

Reply:
I think you need to use [jest-environment-jsdom-global](https://www.npmjs.com/package/jest-environment-jsdom-global) and replace the File object on the global object.

Not sure if it works as it is today with `global.File = MyFileFunc` or `window.File = MyFileFunc` 😕